igcollect/switch.py

```diff
@@ -281,42 +292,15 @@ def get_laggs(snmp, model):
     return None
 
 
-def standarize_portname(port_name):
+def standarize_portname(port_name, model):
     """ Return a Graphite-compatible port name or None if name
         can't be translated
     """
-
-    if re.match('\A(Po|Trk)[0-9]+\Z', port_name):
-        # Dell or HP LAGG
-        return port_name
-    if re.match('\A[0-9]+\Z', port_name):
-        # HP normal port
-        return port_name
-    if re.match('\A[0-9]:[0-9]+\Z', port_name):
-        # Extreme normal port
-        return port_name.replace(':', '_')
-    if re.match('\A(Gi|Te)[0-9]/[0-9]/[0-9]+\Z', port_name):
-        # Dell normal port
-        return port_name.replace('/', '_')
-    if re.match('\A(Fa|Gi|Tu)[0-9]/[0-9]+\Z', port_name):
-        # Cisco ports
-        return port_name.replace('/', '_')
-    if re.match('\A(Fa|Gi|Tu)[0-9]+\Z', port_name):
-        # Cisco tunnels
-        return port_name
-    if re.match('\A(swp)[0-9]+\Z', port_name):
-        # Cumulus ports
-        return port_name
-    g = re.match(
-        '\A(TenGigabitEthernet|fortyGigE) ([0-9]+/[0-9]+)\Z',
-        port_name
-    )
-    if g:
-        # Force 10 MXL port
-        return g.group(2).replace('/', '_')
-    if re.match('\Aethernet[0-9]+/[0-9]+\Z', port_name):
-        # Netiron ports
-        return port_name.replace('/', '_')
+    r = PORT_REGEXP[model].match(port_name)
```

Comment:
What about a key that doesn't exist? KeyError being thrown is ok?

Reply:
It's fine. We only support particular types of switches and the script crashing for any new switch is an indicator for us that SNMP OIDs for the new type must be researched and added.